hypershift-operator/controllers/hostedcluster/internal/platform/agent/agent.go

```diff
@@ -45,13 +54,151 @@ func (p Agent) ReconcileCAPIInfraCR(ctx context.Context, c client.Client, create
 }
 
 func (p Agent) CAPIProviderDeploymentSpec(hcluster *hyperv1.HostedCluster, tokenMinterImage string) (*appsv1.DeploymentSpec, error) {
-	// It is expected that cluster-api-provider-agent and the Infrastructure Operator are installed as a prerequisite
-	return nil, nil
+	deploymentSpec := &appsv1.DeploymentSpec{
+		Replicas: k8sutilspointer.Int32Ptr(1),
+		Selector: &metav1.LabelSelector{
+			MatchLabels: map[string]string{
+				"control-plane": "controller-manager",
```

Comment:
No need to set this labels, they are set by the core controller via `reconcileCAPIProvider`

Reply:
Done